### DISCUSSION TITLE
Spacing between list elements

### DISCUSSION THREAD

**Discussion by wa-rren-dev** [nhsevidence/NICE-Experience#141] [General]
When list text wraps to more than one line it can be hard to distinguish one list item from another. Adding extra spacing between the list items can make it easier to scan through and pick out the individual list items. Not sure of the best way to do this padding? margin? something else?:

Current list:
![image](https://user-images.githubusercontent.com/6341474/81273352-aa731280-9046-11ea-981a-bd9a0a5d1090.png)

With 4px top padding:
![image](https://user-images.githubusercontent.com/6341474/81273392-b4951100-9046-11ea-8a62-9f542417e656.png)

With 8px top padding:
![image](https://user-images.githubusercontent.com/6341474/81273405-ba8af200-9046-11ea-8f66-0a63361516e0.png)

Examples pages: https://www.nice.org.uk/about/covid-19/support-for-developers-of-medicinal-products-for-covid-19

**Comment by ediblecode** [comment]
I'm not sure top padding/margin is the way to go as it moves lists out of alignment with other items potentially, so would favour bottom margin I think. Adding a bottom margin looks like this:

![image](https://user-images.githubusercontent.com/893676/87860235-ab240100-c933-11ea-82ac-e3c4e43fa7ca.png)

Which has the benefit (maybe?) that the first list item of a sub list is closer to the item above. Or does that just look wrong/weird? As in, should it be equal spacing? If we did want equal spacing then we'd have to opt for a padding top I think unless we do something for nested list items too. Not sure if any of that made sense.

**Comment by ediblecode** [comment]
We need to consider tabs and stacked navs if we're changing list item margins/paddings, and ensure there's no regressions in global nav

**Comment by wa-rren-dev** [comment]
> I'm not sure top padding/margin is the way to go as it moves lists out of alignment with other items potentially, so would favour bottom margin I think. Adding a bottom margin looks like this:
> 
> ![image](https://user-images.githubusercontent.com/893676/87860235-ab240100-c933-11ea-82ac-e3c4e43fa7ca.png)
> 
> Which has the benefit (maybe?) that the first list item of a sub list is closer to the item above. Or does that just look wrong/weird? As in, should it be equal spacing? If we did want equal spacing then we'd have to opt for a padding top I think unless we do something for nested list items too. Not sure if any of that made sense.

This example looks good to me, makes sense to group the nested items underneath

**Comment by rnickr** [comment]
Revisiting this as i think this could have a big impact on CKS with it's multiple list structure. Needs a bit more thought as you can get up to 3 level deep lists on CKS but spacing should help make it more scannable and readable until the underlying content can be properly marked up. 

### Examples

Values to be tweaked as needed, ignore left hand nav spacing - this wouldn't change. 


### **3 level list**

**Current**
![image](https://user-images.githubusercontent.com/56125528/91417667-0e655500-e849-11ea-8ea9-b1e6107b928f.png)

**Modified with margin bottom - ul il - 16px,  ul li ul li - 4px, ul li ul li ul li - 0px**
![image](https://user-images.githubusercontent.com/56125528/91417873-52f0f080-e849-11ea-87bb-6999e4076e3d.png)

### **2 level list**
**Current**
![image](https://user-images.githubusercontent.com/56125528/91418156-ad8a4c80-e849-11ea-8d32-d63fca37050b.png)

**Modified with margin bottom - ul li - 16px, ul li ul li - 4px**
![image](https://user-images.githubusercontent.com/56125528/91418696-681a4f00-e84a-11ea-842d-467563229ffd.png)

**Comment by TrishB70** [comment]
The page audit is throwing up a number of examples that could do with this being implemented, for instance https://www.nice.org.uk/about/what-we-do/our-programmes/nice-advice/iapt 
![image](https://user-images.githubusercontent.com/79695728/109318135-635c4500-7845-11eb-984e-503a5a408d89.png)
GOV.UK design system has a nice solution: https://design-system.service.gov.uk/styles/typography/list-spaced/index.html

**Reply by wa-rren-dev** [reply]
@TrishB70 Do you think the [improvements suggested above](https://github.com/nice-digital/nice-design-system/discussions/141#discussioncomment-401771) would do what you need?

**Reply by TrishB70** [reply]
I think a margin-bottom solution similar to the .govuk-list--spaced solution would work. This adds a 15px margin. I wouldn't use padding.

**Reply by rnickr** [reply]
Would we want to keep it spaced using rems rather than pxs?

Couple of alternatives:

0.5rem margin-bottom
![image](https://user-images.githubusercontent.com/56125528/109325621-0ca73900-784e-11eb-8c14-a1bee573041d.png)

1rem margin-bottom
![image](https://user-images.githubusercontent.com/56125528/109327207-fd28ef80-784f-11eb-8c97-7d6991ad60a3.png)

**Reply by TrishB70** [reply]
Yes rems would be more consistent.

**Reply by rnickr** [reply]
@wa-rren-dev - can we get this implemented ? It's been hanging around for a while. There's multiple use cases where this would be an improvement and seems like a relatively easy fix, can this be moved to issues with a view to getting something released?

I can draft up suggestions for the the spacing we would need.

**Reply by wa-rren-dev** [reply]
Sorry this comment was hard to find! Yeah I'll try and slip it in in the next few days. Anything extra you could provide in preparation would be appreciated. Let me know if you want a chat.

**Reply by rnickr** [reply]
yeah nice one, started looking at css that might work but be good to chat through

**Comment by wa-rren-dev** [comment]
Implemented in #175 so locking for now.
